yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,191 @@
+
+from .common import InfoExtractor
+from ..utils import (
+    bug_reports_message,
+    ExtractorError,
+    float_or_none,
+    format_field,
+    int_or_none,
+    traverse_obj,
+    parse_codecs,
+)
+
+
+class AcFunVideoBaseIE(InfoExtractor):
+    def parse_format_and_subtitle(self, video_id, video_info):
+        if 'ksPlayJson' not in video_info:
+            raise ExtractorError(f'Unknown webpage json schema{bug_reports_message()}')
+        playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+
+        fmt_jobj = traverse_obj(playjson, ('adaptationSet', 0, 'representation'))
+        formats = []
+        subtitles = []
+        for video in fmt_jobj:
+            format_list, subtitle_list = self._extract_m3u8_formats_and_subtitles(video['url'], video_id)
+
+            formats += [{
+                **fmt,
+                'fps': int_or_none(video.get('frameRate')),
+                'width': int_or_none(video.get('width')),
+                'height': int_or_none(video.get('height')),
+                'tbr': float_or_none(video.get('avgBitrate')),
+                'ext': 'mp4',
+                'subtitles': subtitle_list,  # it seems AcFun do not have subtitles
+                **parse_codecs(video.get('codecs', ''))
+            } for fmt in format_list]
+
+            subtitles += subtitle_list
+
+        self._sort_formats(formats)
+
+        return {
+            'formats': formats,
+            'subtitles': subtitles,
+            'duration': float_or_none(video_info.get('durationMillis'), 1000),
+            'timestamp': int_or_none(video_info.get('uploadTime'), 1000),
+        }
+
+
+class AcFunVideoIE(AcFunVideoBaseIE):
+    _VALID_URL = r'(?x)https?://(?:www\.acfun\.cn/v/)ac(?P<id>[_\d]+)'
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'info_dict': {
+            'id': '35457073',
+            'title': '1 8 岁 现 状',
+            'thumbnail': 'https://tx-free-imgs.acfun.cn/newUpload/51246077_82bcf86c32c54c4d80cbd624ba4cc38c.jpeg?imageslim',
+            'description': '“赶紧回去！班主任查班了！”',
+            'uploader': '锤子game',
+            'uploader_id': '51246077',
+            'tags': ['电子竞技', 'LOL', 'CF', '搞笑', '真人'],
+            'view_count': 31917,
+            'like_count': 1850,
+            'comment_count': 288,
+            'duration': 174.208,
+            'timestamp': 1656403967
+        },
+        'params': {
+            'skip_download': 'm3u8',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+        json_all = self._search_json(r'window.videoInfo\s*=\s*', webpage, 'videoInfo', video_id)
+
+        if 'currentVideoInfo' not in json_all:
+            raise ExtractorError(f'Unknown webpage json schema{bug_reports_message()}')
```

Comment:
Imo, this error isnt even needed. The below line will raise a good enough error for this situation. But if u want to keep it, do:
```suggestion
            raise ExtractorError('Unknown webpage json schema')
```
bug_report_message is automatically added

Reply:
ok

Reply:
this applies everywhere!

Reply:
> this applies everywhere!
I cleaned other error